tail/tail.go

```diff
@@ -208,12 +218,19 @@ func tailSingleFile(tailer *tail.Tail, file string, stateFile string) chan strin
 
 // tailStdIn is a special case to tail STDIN without any of the
 // fancy stuff that the tail module provides
-func tailStdIn() chan string {
+func tailStdIn(abort chan bool) chan string {
 	lines := make(chan string)
 	input := bufio.NewReader(os.Stdin)
 	go func() {
 		defer close(lines)
+	ReadLines:
 		for {
+			// check for signal triggered exit
+			select {
+			case <-abort:
+				break ReadLines
```

Comment:
because ... 
changed to return, removed the label. :)
(guessing it was a stuck mental pattern from the previous block)